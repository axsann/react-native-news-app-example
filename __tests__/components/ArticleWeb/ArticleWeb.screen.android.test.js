// @flow

import 'react-native'
import React from 'react'
import ArticleWebScreen from '../../../src/components/ArticleWeb/ArticleWeb.screen'
import ShallowRenderer from 'react-test-renderer/shallow'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

jest.unmock('ScrollView')
jest.mock('WebView', () => 'WebView')

jest.mock('Platform', () => {
  const Platform = require.requireActual('Platform')
  Platform.OS = 'android'
  return Platform
})

const navigation = {
  state: {
    params: {
      url: 'http://example.com',
    },
  },
}

describe('ArticleWebScreen', () => {
  it('renders correctly', () => {
    const instance = renderer.create(
      <ArticleWebScreen navigation={navigation} />
    )

    expect(instance.toJSON()).toMatchSnapshot()
  })
})
