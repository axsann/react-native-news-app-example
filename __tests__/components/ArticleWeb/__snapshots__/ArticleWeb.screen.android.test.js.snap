// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleWebScreen renders correctly 1`] = `
<View
  style={
    Object {
      "alignSelf": "stretch",
      "backgroundColor": "transparent",
      "flex": 1,
    }
  }
>
  <WebView
    onError={[Function]}
    onLoad={[Function]}
    source={
      Object {
        "uri": "http://example.com",
      }
    }
    style={
      Object {
        "alignSelf": "stretch",
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "transparent",
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <RNSpinkit
        color="#eeeeee"
        size={100}
        style={
          Array [
            Object {
              "height": 100,
              "width": 100,
            },
            Object {
              "marginTop": -90,
            },
          ]
        }
        type="ThreeBounce"
      />
    </View>
  </View>
</View>
`;
